src/views/student/StudentCourseList.vue

```diff
@@ -1,66 +1,110 @@
+                </div>
+                <div class="btn-tertiary" @click="clearFilter()">Clear filter</div>
+            </div>
+            <hr class="mt-4 mb-8 dark:border-normalgray-700" />
+        </div>
+        <div>
+            <course-list
                 :key="refreshKey"
-                :only-admitted-courses="isMyCoursesPage"
                 :show-all-courses="true"
                 :selected-type="selectedType"
                 :filter="message"
+                @on-updated="matchingCoursesCount = $event"
             />
         </div>
-    </div>
+    </base-view>
 </template>
 
 <script lang="ts">
     import CourseList from "@/components/course/list/student/CourseList.vue";
-    import SeachBar from "@/components/common/SearchBar.vue";
-    import { ref, watch } from "vue";
-    import CourseTypeFilter from "@/components/course/list/common/CourseTypeFilter.vue";
+    import SearchBar from "@/components/common/SearchBar.vue";
+    import { onMounted, ref, watch } from "vue";
+    import Select from "@/components/common/Select.vue";
     import { checkPrivilege } from "@/use/helpers/PermissionHelper";
     import { Role } from "@/entities/Role";
+    import BaseView from "@/views/common/BaseView.vue";
+    import { useStore } from "@/use/store/store";
+    import { useToast } from "@/toast";
 
     export default {
         name: "StudentCourseList",
         components: {
+            BaseView,
             CourseList,
-            SeachBar,
-            CourseTypeFilter,
+            SearchBar,
+            FilterSelect: Select,
         },
         async beforeRouteEnter(_to: any, _from: any, next: any) {
             const response = await checkPrivilege(Role.STUDENT);
-
             if (response.allowed) {
                 return next();
             }
             if (!response.authenticated) {
                 return next("/login");
             }
-
             return next("/redirect");
         },
-
         props: {
             isMyCoursesPage: {
                 type: Boolean,
                 default: false,
             },
         },
         setup(props: any) {
+            const matchingCoursesCount = ref(0);
             let title = ref(props.isMyCoursesPage ? "My Courses" : "Available Courses");
             let message = ref("");
             let refreshKey = ref(false);
 
-            let selectedType = ref("All");
+            const defaultType = "All";
+            let selectedType = ref(defaultType);
+            const types = ref([] as string[]);
+
+            onMounted(async () => {
+                const store = useStore();
+                await store.getters.configuration
+                    .then((config) => {
+                        types.value = [...config.courseTypes];
+                        types.value.unshift("All");
+                    })
+                    .catch((reason) => {
+                        const toast = useToast();
+
+                        toast.error(reason);
```

Comment:
This should not be shown to the user. The reason could only be some API errors (which are already shown via toast from within the handlers) or some runtime error messages, that would be too cryptic for a "normal" user. We could print them to the console or remove it completely

I saw that somewhere else, too.